src/audio/component.c

```diff
@@ -278,26 +281,40 @@ int comp_verify_params(struct comp_dev *dev, uint32_t flag,
 
 		/* set component period frames */
 		component_set_period_frames(dev, buf->stream.rate);
+
+		buffer_unlock(buf);
 	} else {
 		/* for other components we iterate over all downstream buffers
 		 * (for playback) or upstream buffers (for capture).
 		 */
 		buffer_list = comp_buffer_list(dev, dir);
+		clist = buffer_list->next;
+
+		while (clist != buffer_list) {
```

Comment:
can you comment on what we are looking for here

Reply:
I haven't added this loop, just modified it to avoid unnecessary buffer invalidation.